versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/FetchValuesUsingOpenAddressing.java

```diff
@@ -163,4 +167,16 @@ Collection<Key> checkForKeys(
     }
     return keysForNextRound;
   }
+
+  private void checkSegmentIndexValidity(final int segment, final Key key) {
+    // We expect segmentForKey to wrap in case the current round number is large enough to overrun
+    // the segment list, so this shouldn't happen
+    if (keyListsArray.length <= segment) {
+      throw new IllegalStateException(
```

Comment:
Is there any way we can easily trigger this in a test? No worries if not.

Reply:
This was a thought-provoking comment.  The only ways I could induce this exception failure involve modifying this particular file.  After looking at it for a bit, I couldn't do it by limiting myself to only caller behavior, even with an arbitrarily bad/corrupt commit log entry.  Although I like defensive preconditions in general, what I've added here seems like overkill, since any potential violation of this invariant is likely to happen in the same (relatively compact) file where it is checked.  I've just deleted the precondition entirely, relying instead on `segmentForKey`'s new remainder operation with `keyListsArray.length` as divisor.  Though I can certainly bring it back the precondition, if you prefer.